CRM/Member/Form/MembershipRenewal.php

```diff
@@ -787,54 +785,29 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
     // Check and fix the membership if it is STALE
     CRM_Member_BAO_Membership::fixMembershipStatusBeforeRenew($currentMembership, $changeToday);
 
+    // CRM-7297 Membership Upsell - calculate dates based on new membership type
```

Comment:
All of these have been moved outside the if below as they are the same in both places